testtools/matchers/_warnings.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) 2009-2016 testtools developers. See LICENSE for details.
+
+__all__ = [
+    'Warnings',
+    'warning_message',
+    'is_deprecated']
+
+import warnings
+
+from ._basic import Is
+from ._const import Always
+from ._datastructures import MatchesListwise, MatchesStructure
+from ._higherorder import (
+    AfterPreprocessing,
+    Annotate,
+    MatchesAll,
+    Not,
+    )
+from ._impl import Mismatch
+
+
+def warning_message(category_type, message=None, filename=None, lineno=None,
```

Comment:
@thomir After a discussion with @jml yesterday, my understanding is that a large percentage of matcher implementations should just be a function (the matching logic) instead of deriving a new type. Following `ClassName` convention for functions might make the API _appear_ more consistent but it's a bit misleading because it has none of the applications of a function, my 2c would be keeping the conventions for functions and classes separate. (`raises` and `Raises` are possible examples.)